pyro/infer/inspect.py

```diff
@@ -301,10 +311,16 @@ def _get_type_from_frozenname(frozen_name):
         if site["type"] != "sample" or site_is_subsample(site):
             continue
 
+        provenance = get_provenance(
+            site["fn"].log_prob(site["value"])
+            if not site_is_deterministic(site)
+            else site["args"][-1]
+        )
```

Comment:
Using the site value like this will not be correct when `site["fn"].event_dim > 0`.

At `deterministic` sites, the distribution is `Delta(v, event_dim).mask(False)`. The `.mask` messes up the provenance of `log_prob`, but if you use `site["fn"].base_dist` to access the underling `Delta` using `log_prob` should work as expected:
```python
provenance = get_provenance(
    site["fn"].log_prob(msg["value"])
    if not site_is_deterministic(site)
    else site["fn"].base_dist.log_prob(msg["value"])
)
```

Reply:
Understood